### PR TITLE
Charmhub migration

### PR DESCRIPTION
**Pull request by wolsen**
Update the openstack-bundles repository to use charmcraft for bundle publishing and update the stable bundles to release focal-yoga and focal-quincy.

### CODE REVIEW

File: development/ceph-base-focal-quincy/bundle.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  ceph-mon:
+    annotations:
+      gui-x: '750'
+      gui-y: '500'
+    charm: ch:ceph-mon
+    channel: quincy/edge
+    num_units: 3
+    options:
+      expected-osd-count: 3
+      source: distro
```
Comment (freyes) [inline development/ceph-base-focal-quincy/bundle.yaml:23]: focal-quincy should be configuring focal-yoga UCA.

File: development/ceph-base-focal-quincy/bundle.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  ceph-osd:
+    annotations:
+      gui-x: '1000'
+      gui-y: '500'
+    charm: ch:ceph-osd
+    channel: quincy/edge
+    num_units: 3
+    options:
+      osd-devices: /dev/sdb
+      source: distro
```
Comment (freyes) [inline development/ceph-base-focal-quincy/bundle.yaml:37]: focal-quincy should be configuring focal-yoga UCA.

File: stable/ceph-base/bundle.yaml [hunk 3]
```diff
@@ -9,12 +10,13 @@ relations:
...
 series: focal
-services:
+applications:
   ceph-mon:
     annotations:
       gui-x: '750'
       gui-y: '500'
-    charm: cs:ceph-mon-61
+    charm: ch:ceph-mon
+    channel: quincy/edge
```
Comment (freyes) [inline stable/ceph-base/bundle.yaml:19]: `source` should be set to focal-yoga UCA.

File: stable/openstack-base/bundle.yaml [hunk 4]
```diff
@@ -329,15 +342,17 @@ applications:
     annotations:
       gui-x: '300'
       gui-y: '1550'
-    charm: cs:rabbitmq-server-117
+    charm: ch:rabbitmq-server
+    channel: 3.8/stable
```
Comment (freyes) [inline stable/openstack-base/bundle.yaml:346]: we test stable/yoga branches with the 3.9/edge track, hence we should use `3.9/stable` here

File: stable/openstack-telemetry/bundle.yaml [hunk 5]
```diff
@@ -183,35 +183,39 @@ applications:
...
       gui-x: '1500'
       gui-y: '0'
-    charm: cs:aodh-56
+    charm: ch:aodh
     num_units: 1
     options:
       openstack-origin: *openstack-origin
     to:
     - lxd:0
+    channel: yoga/edge
```
Comment (freyes) [inline stable/openstack-telemetry/bundle.yaml:192]: the channels should use the `stable` risk.

**Review by freyes — CHANGES_REQUESTED** [review]
Hi Billy,

There are few things that need to be adjusted, see comments inline.

Best,

File: stable/ceph-base/bundle.yaml [hunk 6]
```diff
@@ -27,7 +29,8 @@ services:
     annotations:
       gui-x: '1000'
       gui-y: '500'
-    charm: cs:ceph-osd-315
+    charm: ch:ceph-osd
+    channel: quincy/edge
```
Comment (pmatulis) [inline stable/ceph-base/bundle.yaml:33]: `quincy/stable`

File: stable/ceph-base/bundle.yaml (hunk 3 above)
Comment (pmatulis) [inline stable/ceph-base/bundle.yaml:19]: `quincy/stable`

**Review by pmatulis — CHANGES_REQUESTED** [review]
Use `quincy/stable` as channel for the two Ceph charms.

File: development/ceph-base-focal-quincy/bundle.yaml (hunk 2 above)
Comment (wolsen) [reply development/ceph-base-focal-quincy/bundle.yaml:37]: ack

File: development/ceph-base-focal-quincy/bundle.yaml (hunk 1 above)
Comment (wolsen) [reply development/ceph-base-focal-quincy/bundle.yaml:23]: ack

File: stable/ceph-base/bundle.yaml (hunk 3 above)
Comment (wolsen) [reply stable/ceph-base/bundle.yaml:19]: ack

File: stable/openstack-base/bundle.yaml (hunk 4 above)
Comment (wolsen) [reply stable/openstack-base/bundle.yaml:346]: ack

File: stable/ceph-base/bundle.yaml (hunk 6 above)
Comment (wolsen) [reply stable/ceph-base/bundle.yaml:33]: ack

File: stable/ceph-base/bundle.yaml (hunk 3 above)
Comment (wolsen) [reply stable/ceph-base/bundle.yaml:19]: ack

**Review by freyes — APPROVED** [review]
LGTM.

**Pull request merged**